source/MaterialXCore/Document.h

```diff
@@ -78,13 +79,13 @@ class MX_CORE_API Document : public GraphElement
     /// Return the NodeGraph, if any, with the given name.
     NodeGraphPtr getNodeGraph(const string& name) const
     {
-        return getChildOfType<NodeGraph>(name);
+        return hasDataLibrary() ? getChildofTypeFromLibrary<NodeGraph>(getDataLibrary(), name) : getChildOfType<NodeGraph>(name);
```

Comment:
@ashwinbhat For this category of updated method implementation (e.g. `getNodeGraph`, `getNodeDef`, `getAttributeDef`), I believe we actually need to check the content document when the data library _has no matching child_, not merely in the case when the data library is not present at all.

Does that sound right to you as well?

Reply:
@jstone-lucasfilm I think the implementation of `template <class T> shared_ptr<T> getChildofTypeFromLibrary` will fallback to get this behavior you are suggesting.

Reply:
Ah, thanks for that clarification, @ashwinbhat!

Now I see the pattern that you're proposing here, and one suggestion that I would make would be to handle the `nullptr` data library case _inside_ of `getChildofTypeFromLibrary`, allowing you to write the external call in a clearer way:

```
return getChildofTypeFromLibrary<NodeGraph>(getDataLibrary(), name);
```

We might even consider renaming the function from `getChildofTypeFromLibrary` to `getChildOfTypeWithLibrary`, in order to make this pattern as clear as possible to readers.